config/src/main/java/org/axonframework/config/DefaultConfigurer.java

```diff
@@ -645,6 +648,15 @@ public Configurer registerHandlerDefinition(
         return this;
     }
 
+    @Override
+    public Configurer configureLifecyclePhaseTimeout(long timeout, TimeUnit timeUnit) {
+        assertStrictPositive(timeout, "The lifecycle phase timeout should be strictly positive");
```

Comment:
Should we care about a minimum timeout here?
For example the default 5s

Reply:
I was in doubt about this. It's hard for us to come up with a value that's the bare minimum. What if somebody has a very small application with only a single infra component? Then, 5 seconds would be too much.
But, what if they have a ton of infra components, adding their own into the lifecycle phases too? Then, 5 seconds could be too little.

So, as I feel a lot of uncertainty here, I tried to cover this concern with the JavaDoc.
Axon will throw clear warning signs if phases can't be executed.
Furthermore, if that happens, people should expect issues.
I figured that was the most lenient solution instead of hitting a hard minimum other than strictly positive.